### ISSUE TITLE
Group and/or name logical switches and special functions

### ISSUE THREAD

**Issue by Daannoa** [EdgeTX/edgetx#1896]
### Is there an existing issue for this feature request?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

I use logical switches to perform different tasks.
Each task may require several logic switches.
It then becomes difficult to know which logical switches are used for which task.
The situation is the same for the special functions

### Describe the solution you'd like

Would it be possible to add the ability to group logical switches ? ( same question for the special fonctions .)

this gives us the ability to group logical switches that are participating in the same task.

### Describe alternatives you've considered

An alternative would be to allow naming of logic switches and special functions.
(like INPUTS with their "Input name")

**Comment by NemesisXB** [comment]
I like this. I would take it one step further in saying that you should be able to group logical switches, special funtions and global variables, as I have functions that require all three of these to accomplish the required task.

**Comment by pfeerick** [comment]
I think a combination of both are needed... SF/LS need to be named and also comments/notes for each, and also grouping.

**Comment by elecpower** [comment]
Whilst I am in support remember our B&W targets memory limits

**Comment by pfeerick** [comment]
Yeah, B&W may be unfortunate one and not get this... due to memory AND firmware size constraints...

**Comment by Neutral67fr** [comment]
There are many lines available for LS and SF... So I use to group the lines by functions, starting each line multiple of 10.
 e.g. LS01 to LS05 are for FM1 to FM5, LS10 to LS19 (or more if needed) are for SF triggering, etc...
  It's maybe not so convenient than names, but offer a standard organization (that allows copy/paste between models). But anyway naming limited to 3 char. is already a hassle for everything else (inputs, mixes, GV, curves...) so its doesn't help much anyway ;-)

**Comment by c79d82** [comment]
I have encountered this exact organisational problem while moving a model from a RadioMaster GX12 to a TX16S Mk3 MAX.

On the GX12, Logical Switches are displayed in their fixed numbered positions, including unused slots. I deliberately left an unused Logical Switch between groups of related switches, which produced a clearly visible blank line between functional blocks.

For example:

* L01–L04: pre-arm logic
* L05: unused separator
* L06–L08: arming-state announcements
* L09: unused separator
* L10–L13: Turtle Mode logic

On the TX16S colour-screen interface, unused Logical Switch slots are hidden. The remaining switches retain their original numbers, but all visible rows are placed directly beneath one another, so the visual grouping is lost.

Custom names and proper groups would be the ideal solution. For colour-screen radios, optional colours or coloured group headings could also make complex configurations much easier to understand.

A potentially smaller interim improvement would be an option such as:

**Show unused Logical Switch slots**

That would restore the useful fixed-list behaviour without requiring additional model-data fields. Users could once again leave an empty slot as a visual separator, although it does sacrifice a usable Logical Switch.

Possible solutions, from simplest to most comprehensive:

1. Optional display of unused Logical Switch slots.
2. User-defined names for individual Logical Switches.
3. Named groups or section headings containing several Logical Switches.
4. Optional colours for Logical Switches or groups on colour-screen radios.

The main problem is not merely remembering what one Logical Switch does; it is recognising that several adjacent Logical Switches together implement one larger function.
